catalog/spatial/registry/registry-admin-modules/registry-admin-local-ui/src/main/webapp/js/model/Segment.js

```diff
@@ -61,43 +73,45 @@ define([
                     type: entry.type,
                     isSlot: true,
                     required: entry.required ? entry.required : false,
+                    advanced: entry.advanced,
                     multiValued: entry.multiValued ? entry.multiValued : false,
+                    inlineGroup: entry.inlineGroup,
                     min: entry.min,
                     max: entry.max,
                     regex: entry.regex,
                     regexMessage: entry.regexMessage,
                     possibleValues: entry.possibleValues,
                     editable: entry.editable,
-                    parentId: segId
+                    parentId: seg.get('segmentId')
                 });
                 var slotFound = false;
                 _.each(backingData.Slot, function (slot) {
                     if (slot.name === name) {
                         slotFound = true;
                         var values = getSlotValue(slot, entry.type, entry.multiValued);
                         if(!values || (_.isArray(values) && values.length === 0)) {
-                            if(entry.values){
-                                if(_.isArray(entry.values)){
-                                    values = entry.values.slice(0);
+                            if(entry.value){
+                                if(_.isArray(entry.value)){
+                                    values = entry.value.slice(0);
                                 } else {
-                                    values = entry.values;
+                                    values = entry.value;
                                 }
                             }
                         }
                         addValueFields(field, values);
                         addedSlots.push(name);
                     }
                 });
-                if(!slotFound && entry.values){
+                if(!slotFound && entry.value){
                     var values = [];
-                    if(_.isArray(entry.values)){
-                        values = entry.values.slice(0);
+                    if(_.isArray(entry.value)){
+                        values = entry.value.slice(0);
                     } else {
-                        values = entry.values;
+                        values = entry.value;
                     }
                     addValueFields(field, values);
                 }
-                field.on('change', field.validate, field);
+                field.setupChangeListener();
```

Comment:
Why not call `setupChangeListener` in the `Field` initializer instead? Are you trying to postpone it until the field is fully initialized or something?